File: backend/src/entities/Note.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, DeleteDateColumn, ManyToOne } from "typeorm";
import { Company } from "./Company";

@Entity()
export class Note {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column()
  title!: string;

  @Column({ type: "text", nullable: true })
  body?: string;

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;

  @DeleteDateColumn()
  deletedAt?: Date;

  @ManyToOne(() => Company, (company) => company.notes)
  company?: Company;
}

File: server/app/models/note.py
from typing import Optional
from beanie import Document, Indexed
from datetime import datetime

class Note(Document):
    title: Optional[str] = None
    content: str
    is_pinned: bool = False
    related_to_type: str # company, deal, lead, person, task, product
    related_to_id: str # The ID of the related entity as string
    created_by: Optional[str] = None # User UUID
    created_at: datetime = datetime.utcnow()
    updated_at: datetime = datetime.utcnow()

    class Settings:
        name = "notes"
